apps/worker/tasks/dlq_recovery.py

```diff
@@ -0,0 +1,241 @@
+"""
+Dead Letter Queue (DLQ) Recovery Task
+
+This task allows manual recovery of tasks that were saved to the DLQ after
+exhausting all retries. Tasks can be re-queued for processing after investigation.
+"""
+
+import logging
+from typing import Any
+
+import orjson
+
+from app import celery_app
+from shared.celery_config import (
+    DLQ_KEY_PREFIX,
+    dlq_recovery_task_name,
+    upload_task_name,
+)
+from shared.helpers.redis import get_redis_connection
+from tasks.base import BaseCodecovTask
+
+log = logging.getLogger(__name__)
+
+
+class DLQRecoveryTask(BaseCodecovTask, name=dlq_recovery_task_name):
+    """
+    Task to recover tasks from Dead Letter Queue.
+
+    This task allows operators to:
+    1. List tasks in DLQ
+    2. Re-queue specific tasks for retry
+    3. Delete tasks from DLQ after recovery
+
+    Args:
+        dlq_key: Full DLQ key (e.g., "task_dlq/app.tasks.upload.Upload/{repoid}/{commitid}")
+        action: Action to perform - "list", "recover", or "delete"
+        task_name_filter: Optional filter to only show/recover tasks matching this name
+    """
+
+    def run_impl(
+        self,
+        db_session,
+        dlq_key: str | None = None,
+        action: str = "list",
+        task_name_filter: str | None = None,
+        **kwargs,
+    ) -> dict[str, Any]:
+        """
+        Run DLQ recovery operations.
+
+        Actions:
+        - "list": List all DLQ keys (optionally filtered by task_name_filter)
+        - "recover": Re-queue tasks from a specific DLQ key
+        - "delete": Delete tasks from a specific DLQ key
+        """
+        redis_conn = get_redis_connection()
+
+        if action == "list":
+            return self._list_dlq_keys(redis_conn, task_name_filter)
+        elif action == "recover":
+            if not dlq_key:
+                return {
+                    "success": False,
+                    "error": "dlq_key required for recover action",
+                }
+            return self._recover_tasks(redis_conn, dlq_key)
+        elif action == "delete":
+            if not dlq_key:
```

Comment:
I think all of these CRUD operations don't make sense per se, I think it's worth considering making dead letter tasks into an object to avoid this work.